client/components/SurrenderForm.js

```diff
@@ -0,0 +1,170 @@
+        errorsArray.push(fieldName)
+      }
+      if (surrenderData[fieldName] === "Select Pet Type")
+        errorsArray.push(fieldName);
+      else if (surrenderData[fieldName] === "Choose")
+        errorsArray.push(fieldName);
+    })
+    return errorsArray
+  }
+
+
+  const onChange = event => {
+    setSurrenderData({
+      ...surrenderData, [event.currentTarget.name]: event.currentTarget.value
+    })
+  }
+
+  const clearForm = () => {
+    setSurrenderData({
+      name: "",
+      phoneNumber: "",
+      email: "",
+      petName: "",
+      petAge: "",
+      petType: "Select Pet Type",
+      petImage: "",
+      vaccinationStatus: "No",
+    });
+  }
+
+  const addSurrenderInfo = async () => {
+    try {
+      const response = await fetch("/api/v1/surrender", {
+        method: 'POST',
+        headers: new Headers({
+          'Content-Type': 'application/json'
+        }),
+        body: JSON.stringify(surrenderData)
+      })
+      if (!response.ok) {
+        const errorMessage = `${response.status} (${response.statusText})`;
+        const error = new Error(errorMessage);
+        throw (error);
+      }
+      const body = await response.json();
+      setPosted(true);
+    } catch (err) {
+      console.error(`Error in fetch: ${err.status} (${err.message})`);
+    }
+  }
+
+  const onSubmit = async event => {
+    event.preventDefault();
+    let errorsArray = checkForErrors();
+    if (errorsArray.length === 0) {
+      await addSurrenderInfo();
+      clearForm();
+      setErrors([]);
+    } else {
+      setErrors(errorsArray);
+    }
+  }
+
+  const message = posted ? "Your surrender request is in process." : "";
+  return (
+    <div>
+      <div>{errorContainer}</div>
+      <h2>Pet Surrender Application</h2>
+      <h3>{message}</h3>
+      <form onSubmit={onSubmit}>
+        <label htmlFor="name" >Name:
+          <input
+            id="name"
+            type="text"
+            name="name"
+            value={surrenderData.name}
+            onChange={onChange} />
+        </label>
+        <label htmlFor="phone-number">Phone Number:
+          <input
+            id="phone-number"
+            type="text"
+            name="phoneNumber"
+            value={surrenderData.phoneNumber}
+            onChange={onChange} />
+        </label>
+        <label htmlFor="email">Email:
+          <input
+            id="email"
+            type="text"
+            name="email"
+            value={surrenderData.email}
+            onChange={onChange} />
+        </label>
+        <label htmlFor="petType">Pet Type
+          <select name="petType" id="petType" onChange={onChange} value={surrenderData.petType}>
+            <option value="Select Pet Type">Select Pet type</option>
+            <option value="dog">dog</option>
+            <option value="cat">cat</option>
+            <option value="hamster">hamster</option>
+            <option value="fish">fish</option>
+            <option value="turle">turtle</option>
```

Comment:
Instead of hardcoding here we can query the database for these pet types.  We will need to fire off a fetch request that should return with all the pet types stored in the database.  Then we can store that list in state and map over it to create these `<option>` elements dynamically

Reply:
Well, that was several hours fixing things implementing this broke.